src/App.jsx

```diff
@@ -14,45 +16,112 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+export const App = () => {
+  const [visibleGoods, setVisibleGoods] = useState(goodsFromServer);
+  const [sortFiled, setSortFiled] = useState('');
+  const [reverse, setReverse] = useState('');
+
+  function sortAlphabetically() {
+    if (reverse === 'R') {
+      setVisibleGoods([...visibleGoods].sort().reverse());
+    } else {
+      setVisibleGoods([...visibleGoods].sort());
+    }
+
+    setSortFiled('A');
+  }
+
+  function sortByLength() {
+    if (reverse === 'R') {
+      setVisibleGoods([...visibleGoods].sort(
+        (good1, good2) => good2.length - good1.length,
+      ));
+    } else {
+      setVisibleGoods([...visibleGoods].sort(
+        (good1, good2) => good1.length - good2.length,
+      ));
+    }
+
+    setSortFiled('L');
+  }
+
+  function reverseSort() {
+    setVisibleGoods([...visibleGoods].reverse());
+    if (reverse === '') {
+      setReverse('R');
+    } else {
+      setReverse('');
+    }
+  }
 
```

Comment:
Consider joining the logic of 3 functions into one function with a switch case for sort type, and an if statement to check reverse and reversing an array. It will help to decrease repeating code here

Reply:
but how can i do it here? 
![Uploading Screenshot 2023-11-02 at 17.44.14.png…]()

Reply:
<img width="704" alt="Screenshot 2023-11-02 at 17 44 14" src="https://github.com/mate-academy/react_list-of-goods-js/assets/112933414/8bd269f9-e060-4606-b1aa-3fec67d9050a">

Reply:
You should use sort methods with parameters and make it in one function. You can set switch statement inside of sort method, and select which one of sorts you should use at the moment. And after sorting you can check with "if" should you reverse the array or not.
For the future, please ask questions in chat, the mentor doesn't get notifications triggered by your answer on github